### DISCUSSION TITLE
What is the use for default host?

### DISCUSSION THREAD

**Discussion by sandrodz** [nginx-proxy/nginx-proxy#1963] [Q&A] [answered]
<img width="921" alt="screen shot 2017-10-17 at 1 24 23 pm" src="https://user-images.githubusercontent.com/8479569/31657109-9e095924-b33e-11e7-9b91-e2fdbbbe4048.png">

**Comment by kamermans** [comment]
It is to tell `nginx-proxy` what site to serve for requests that have no `Host` header or if there are no sites that match the given `Host` header.

**Comment by sandrodz** [comment]
In that case it doesn't seem to work.

I get 503 Service Temporarily Unavailable instead of default host.

**Comment by kamermans** [comment, marked as answer]
Here is an example that shows how it's supposed to work:

First, without `DEFAULT_HOST`:
```
# Start nginx-proxy in one window
docker run -ti --rm -p 80:80 -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy

# In another window start this hello world container
docker run -ti --rm -e VIRTUAL_HOST=foo.com tutum/hello-world

# In a third window run these commands
# Send a request to nginx-proxy for foo.com
curl -I --resolve foo.com:80:127.0.0.1 http://foo.com/
# You will see "HTTP/1.1 200 OK" as expected because the hostname matches

# Send a request to nginx-proxy for bar.com
curl -I --resolve bar.com:80:127.0.0.1 http://bar.com/
# You will see "HTTP/1.1 503 Service Temporarily Unavailable" because no VIRTUAL_HOST matches "bar.com" and there is no DEFAULT_HOST
```

Now, the same test with `DEFAULT_HOST`:
```
# Start nginx-proxy in one window
docker run -ti --rm -p 80:80 -e DEFAULT_HOST=foo.com -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy

# In another window start this hello world container
docker run -ti --rm -e VIRTUAL_HOST=foo.com tutum/hello-world

# In a third window run these commands
# Send a request to nginx-proxy for foo.com
curl -I --resolve foo.com:80:127.0.0.1 http://foo.com/
# You will see "HTTP/1.1 200 OK" as expected because the hostname matches

# Send a request to nginx-proxy for bar.com
curl -I --resolve bar.com:80:127.0.0.1 http://bar.com/
# You will see "HTTP/1.1 200 OK" because although no VIRTUAL_HOST matches "bar.com", the DEFAULT_HOST directive told nginx-proxy to serve the content for "foo.com" in this case
```

Does this clear things up?  If this is what you expect and it's still not working, then you should bash into the `nginx-proxy` container and check your `/etc/nginx/conf.d/default.conf`.  If `DEFAULT_HOST` matches one of the `VIRTUAL_HOST` directives from a running container, you will see `default_server` in the `listen` statement of the `location` for that host:

```
server {
	server_name foo.com;
	listen 80 default_server;
	access_log /var/log/nginx/access.log vhost;
	location / {
		proxy_pass http://foo.com;
	}
}
```

If you don't see `default_server` in that file, then no `VIRTUAL_HOST` matched the `DEFAULT_HOST` you specified.

**Comment by v-stickykeys** [comment]
I set my DEFAULT_HOST=localhost but when I curl localhost I am getting the nginx 503.

Also default.conf has `listen 80 ;`
